scripts/build-ancillary-repository.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+TOP=$(git rev-parse --show-toplevel 2>/dev/null)
+
+if [[ -z "$TOP" ]]; then
+	echo "Must be run inside the git repsitory." 2>&1
+	exit 1
+fi
+
+. "$TOP/scripts/functions.sh"
+
+set -o xtrace
+set -o errexit
+set -o pipefail
+
+function resolve_s3_uri()
+{
+	local PKG_URI="$1"
```

Comment:
[nit] we should generally use lower case for local variables (not sure if our style checks complain though)

Reply:
fair enough.. I don't like the look of it, but I know it's pretty standard to do ti that way.. just too closely resemble camel case and perl for my taste :p  I'll update it.